### PR TITLE
SAMA5D2 TDC, ADC and TC Corrections

### PR DESCRIPTION
**Pull request by TimJTi**
## Summary
The feedback is still coming!
In double checking it all, I found an error in sam_tc.c as (perhaps some other change meant) "int ret" had not been declared.

## Impact
None

## Testing
Re run my SAMA5D2 board TSD and ADC tests

Leaving as a draft as I expect more things will be found.

### CODE REVIEW

File: arch/arm/src/sama5/sam_adc.c [hunk 1]
```diff
@@ -1890,7 +1889,7 @@ static void sam_adc_gain(struct sam_adc_s *priv)
   /* Set GAIN0 only.  GAIN0 will be used for all channels. */
 
   sam_adc_putreg(priv, SAM_ADC_CGR, ADC_CGR_GAIN0(CONFIG_SAMA5_ADC_GAIN));
-#endif /* CONFIG_SAMA5_ADC_ANARCH */
+   #endif /* CONFIG_SAMA5_ADC_ANARCH */
```
Comment (pkarashchenko) [inline arch/arm/src/sama5/sam_adc.c:1892]:
```suggestion
#  endif /* CONFIG_SAMA5_ADC_ANARCH */
```

File: arch/arm/src/sama5/sam_tc.c [hunk 2]
```diff
@@ -1451,7 +1452,8 @@ int sam_tc_divisor(uint32_t frequency, uint32_t *div, uint32_t *tcclks)
 
   if (tcclks)
     {
-      tmrinfo("return tcclks=%08lx\n", (unsigned long)TC_CMR_TCCLKS(ndx));
+      tmrinfo("return tcclks=%08" PRIx32 "\n",
```
Comment (pkarashchenko) [inline arch/arm/src/sama5/sam_tc.c:1455]: This place does not need PRIx32 macro since argument is casted to unsigned long

Comment (TimJTi) [reply arch/arm/src/sama5/sam_tc.c:1455]: seems very inconsistent to me to sometimes recast and sometimes use a macro?

Comment (pkarashchenko) [reply arch/arm/src/sama5/sam_tc.c:1455]: All is consistent. `uint32_t`/`int32_t` should be printed with `PRId/u/x32` macro while `long`/`unsigned long` should use `ld/u/x` and `int`/`unsigned int` with `d/u/x`

**Comment by TimJTi** [comment]
checkpatch.sh reporting things that look OK to me. Can't see the wood for the trees at the moment...

File: arch/arm/src/sama5/sam_tc.c [hunk 3]
```diff
@@ -1263,11 +1264,11 @@ void sam_tc_setregister(TC_HANDLE handle, int regid, uint32_t regval)
 
   DEBUGASSERT(chan && regid < TC_NREGISTERS);
 
-  tmrinfo("Channel %d: Set register RC%d to %08lx\n",
-          chan->chan, regid, (unsigned long)regval);
+  tmrinfo("Channel %d: Set register RC%d to %08lx"\n",
+           chan->chan, regid, (unsigned long)regval);
```
Comment (pkarashchenko) [inline arch/arm/src/sama5/sam_tc.c:1268]:
```suggestion
          chan->chan, regid, (unsigned long)regval);
```

File: arch/arm/src/sama5/sam_tc.c [hunk 4]
```diff
@@ -1263,11 +1264,11 @@ void sam_tc_setregister(TC_HANDLE handle, int regid, uint32_t regval)
 
   DEBUGASSERT(chan && regid < TC_NREGISTERS);
 
-  tmrinfo("Channel %d: Set register RC%d to %08lx\n",
-          chan->chan, regid, (unsigned long)regval);
+  tmrinfo("Channel %d: Set register RC%d to %08lx"\n",
```
Comment (pkarashchenko) [inline arch/arm/src/sama5/sam_tc.c:1267]:
```suggestion
  tmrinfo("Channel %d: Set register RC%d to %08lx\n",
```

File: arch/arm/src/sama5/sam_tc.c [hunk 5]
```diff
@@ -1443,19 +1445,19 @@ int sam_tc_divisor(uint32_t frequency, uint32_t *div, uint32_t *tcclks)
   if (div)
     {
       uint32_t value = sam_tc_freqdiv_lookup(ftcin, ndx);
-      tmrinfo("return div=%lu\n", (unsigned long)value);
+      tmrinfo("return div=%lu\n", (unsigned long) value);
```
Comment (pkarashchenko) [inline arch/arm/src/sama5/sam_tc.c:1448]:
```suggestion
      tmrinfo("return div=%lu\n", (unsigned long)value);
```

File: arch/arm/src/sama5/sam_tc.c [hunk 6]
```diff
@@ -1443,19 +1445,19 @@ int sam_tc_divisor(uint32_t frequency, uint32_t *div, uint32_t *tcclks)
...
       *div = value;
     }
 
   /* Return the TCCLKS selection */
 
   if (tcclks)
     {
-      tmrinfo("return tcclks=%08lx\n", (unsigned long)TC_CMR_TCCLKS(ndx));
+      tmrinfo("return tcclks=%08lx\n", (unsigned long) TC_CMR_TCCLKS(ndx));
```
Comment (pkarashchenko) [inline arch/arm/src/sama5/sam_tc.c:1456]:
```suggestion
      tmrinfo("return tcclks=%08lx\n", (unsigned long)TC_CMR_TCCLKS(ndx));
```

Comment (TimJTi) [reply arch/arm/src/sama5/sam_tc.c:1456]: I put the space in because that was the only way checkpatch.sh was happy and certainly not what i would usually do! I will try again tomorrow as I think the script was having a bad day (which isn't even possible).

File: arch/arm/src/sama5/sam_tc.c (hunk 5 above)
Comment (TimJTi) [reply arch/arm/src/sama5/sam_tc.c:1448]: I put the space in because that was the only way checkpatch.sh was happy and certainly not what i would usually do! I will try again tomorrow as I think the script was having a bad day (which isn't even possible).

**Comment by TimJTi** [comment]
Note to self....do not try and quickly push supposedly trivial changes :(

**Comment by TimJTi** [comment]
OK...so in trying to fix someone else's code I found I didn't have any timer/counters enabled in my config, this attempt So the code was not actually being compiled, so wasn't picking up a typo....which then caused checkpatch.sh to do strange things based that earlier error I hadn't corrected (but allowed the build to succeed). Nightmare.

So there's a lesson here...somewhere....just don't know what as yet!

But, maybe, just maybe, this patch on a patch will be OK now. Who knows :(

File: arch/arm/src/sama5/sam_tc.c [hunk 7]
```diff
@@ -1458,4 +1461,4 @@ int sam_tc_divisor(uint32_t frequency, uint32_t *div, uint32_t *tcclks)
   return OK;
 }
 
-#endif /* CONFIG_SAMA5_TC0 || CONFIG_SAMA5_TC1 || CONFIG_SAMA5_TC2 */
+#endif /* CONFIG_SAMA5_TC0 || CONFIG_SAMA5_TC1 || CONFIG_SAMA5_TC2 */
```
Comment (xiaoxiang781216) [inline arch/arm/src/sama5/sam_tc.c:1464]: revert the change?

File: arch/arm/src/sama5/sam_adc.c (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/arm/src/sama5/sam_adc.c:1892]: not fix yet.

File: arch/arm/src/sama5/sam_tc.c (hunk 7 above)
Comment (TimJTi) [reply arch/arm/src/sama5/sam_tc.c:1464]: Not sure what I changed or you want reverted?

File: arch/arm/src/sama5/sam_adc.c (hunk 1 above)
Comment (TimJTi) [reply arch/arm/src/sama5/sam_adc.c:1892]: I reverted all my changes that included this fix when I was trying to work out why checkpatch.sh was reporting so many what seemed to be daft errors...forgot to put this one back in as i was concentrating on sam_tc.c

I'll double check I did include all the other suggestions.

File: arch/arm/src/sama5/sam_tc.c (hunk 7 above)
Comment (TimJTi) [reply arch/arm/src/sama5/sam_tc.c:1464]: Oh - CR missing at the end. No idea how that happened...although it built Ok and passed checkpatch.sh?

File: arch/arm/src/sama5/sam_adc.c [hunk 8]
```diff
@@ -1890,7 +1889,7 @@ static void sam_adc_gain(struct sam_adc_s *priv)
   /* Set GAIN0 only.  GAIN0 will be used for all channels. */
 
   sam_adc_putreg(priv, SAM_ADC_CGR, ADC_CGR_GAIN0(CONFIG_SAMA5_ADC_GAIN));
-#endif /* CONFIG_SAMA5_ADC_ANARCH */
+  #endif /* CONFIG_SAMA5_ADC_ANARCH */
```
Comment (xiaoxiang781216) [inline arch/arm/src/sama5/sam_adc.c:1892]: @TimJTi the change should be?
```suggestion
#  endif /* CONFIG_SAMA5_ADC_ANARCH */
```

**Comment by xiaoxiang781216** [comment]
let's squash to one patch and update the commit message to reflect the real channge.

**Comment by xiaoxiang781216** [comment]
Yes, the title “Correction” is too general, it's better to add prefix to indicate the change is for SAMA5D2 and fix the printf formatter.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]
